cloud-sql/mysql/Startup.cs

```diff
@@ -55,34 +54,100 @@ public void ConfigureServices(IServiceCollection services)
                     options.Version = "Test";
                 });
             }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddScoped(typeof(MySqlConnectionStringBuilder),
+                (IServiceProvider) => GetMySqlConnectionString());
+            services.AddScoped<MySqlConnect>();
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
             _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        public class MySqlConnect
         {
-            DbConnection connection;
+            public DbConnection GetMySqlConnection(
+                MySqlConnectionStringBuilder connectionString)
+            {
+                // [START cloud_sql_mysql_dotnet_ado_backoff]
+                var connection = Policy
+                    .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                    .WaitAndRetry(new[]
+                    {
+                    TimeSpan.FromSeconds(1),
+                    TimeSpan.FromSeconds(2),
+                    TimeSpan.FromSeconds(5)
+                    }, (result, timeSpan, retryCount, context) =>
+                    {
+                        // Log any warnings here.
+                    })
+                    .Execute(() =>
+                    {
+                        // Return a new connection.
+                        // [START_EXCLUDE]
+                        if (Environment.GetEnvironmentVariable("DB_HOST") != null)
+                        {
+                            return NewMysqlTCPConnection(connectionString);
+                        }
+                        else
+                        {
+                            return NewMysqlUnixSocketConnection(connectionString);
+                        }
+                        // [END_EXCLUDE]

```

Comment:
I think you can do here as follows, because the connection string already knows if this is a socket or a tcp connection. And you wouldn't need to exclude it  from the snippet.
This applies to the three examples.
```suggestion
                        DbConnection connection = new MySqlConnection(connectionString.ConnectionString);

```

Reply:
Done.